app/services/inline_uploads.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require_dependency "pretty_text"
+
+class InlineUploads
+  def self.process(markdown)
+    cooked_fragment = Nokogiri::HTML::fragment(PrettyText.cook(markdown.dup))
+    link_occurences = []
+
+    cooked_fragment.traverse do |node|
+      if node.name == "img"
+        # Do nothing
+      elsif !(node.children.count == 1 && node.children[0].children.blank?)
+        next
+      end
+
+      if seen_link = matched_uploads(node).first
```

Comment:
Shouldn't we call `matched_uploads(node)` on `a` and `img` nodes only?

Reply:
Nope we need to walk all the nodes because there are cases like code blocks which may include a link.